dune-precice/dune-precice/couplinginterface.hh

```diff
@@ -0,0 +1,264 @@
+// -*- tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+// vi: set et ts=4 sw=2 sts=2:
+
+#ifndef DUNE_PRECICE_HH
+#define DUNE_PRECICE_HH
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <dune/istl/io.hh>
+#include <dune/istl/bvector.hh>
+
+#include <dune/functions/functionspacebases/interpolate.hh>
+
+#include <precice/SolverInterface.hpp>
```

Comment:
Does it make sense to include it inside the `Dune::preCICE` namespace to keep the scope of preCICE more narrow?

Reply:
I'm not sure, does it make a big difference?

Reply:
I am not sure what would be the best here. Including it within the namespace would hide preCICE and its `precice` namespace a bit more from users. From my perspective this would be something good since we want people to use the adapter. However, I did not find any examples regarding good and bad practices. This is for sure a small thing we can also easily change later. I just wanted to mention it.

Maybe @fsimonis has an idea/suggestion regarding the best practices regarding includes inside/outside namespaces?!

Reply:
I think we can just keep this way, it is also more clear from an outside view (at least in my opinion), because you have all headers before starting to do anything.